File: _0003.java
public class _0003 {
    public static void main(String[] argv) {
	long number = 600851475143L;
	int divisor = 2;
	while (number > 1) {
	    if (0 == (number % divisor)) {
		number /= divisor;
	    } else {
		divisor++;
	    }
	}
	System.out.println(divisor);
    }
}

File: 0003.cpp
#include <iostream>
#include <vector>
#include <functional>

int fact(long int num) {
  int divisor = 2;
  while (num > 1) {
    if (0 == (num % divisor)) {
      num /= divisor;
    } else {
      divisor++;
    }
  }
  return divisor;
}

void fn0() {
  //std::cout << fact(13195) << std::endl;
  std::cout << fact(600851475143) << std::endl;
}

std::vector<std::function<void()>> progs = { fn0 };
